python/tk_desktop/desktop_window.py

```diff
@@ -1227,82 +1262,48 @@ def __launch_app_proxy_for_project(self, project, requested_pipeline_configurati
         # From this point on, we don't touch the UI anymore.
         self.project_overlay.start_progress()
 
-        self._sync_thread = ProjectSynchronizationThread(toolkit_manager, project)
```

Comment:
No more need for the sync thread, since everything is synced in the background process.